src/screens/internal/Anomalies/AnomaliesSelector.tsx

```diff
@@ -0,0 +1,160 @@
+import get from 'lodash/get';
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router';
+import { useHistory } from 'react-router-dom';
+import { fail } from 'common/utils';
+import * as QueryString from 'query-string';
+import { OptionsSelectorWrapper } from 'screens/internal/CompareSnapshots/OptionsSelector.style';
+import {
+  FormControl,
+  InputLabel,
+  MenuItem,
+  TextField,
+} from '@material-ui/core';
+import {
+  useMainSnapshot,
+  getNumericParamValue,
+  Select,
+} from '../CompareSnapshots/OptionsSelector';
+import {
+  AnnotationType,
+  ANNOTATION_TYPES,
+  annotationTypeNames,
+  AnnotationSelectorProps,
+  AnnotationSelectorInnerProps,
+} from './utils';
+import regions from 'common/regions';
+
+export function AnnotationsSelector(props: AnnotationSelectorProps) {
+  const mainSnapshot = useMainSnapshot();
+  if (!mainSnapshot) {
+    return null;
+  } else {
+    return <AnnotationsSelectorInner mainSnapshot={mainSnapshot} {...props} />;
+  }
+}
+
+function AnnotationsSelectorInner({
+  mainSnapshot,
+  onNewOptions,
+}: AnnotationSelectorInnerProps) {
+  const location = useLocation();
+  const history = useHistory();
+
+  const params = QueryString.parse(history.location.search);
+
+  const [snapshot, setSnapshot] = useState(
+    getNumericParamValue(params, 'snapshot', mainSnapshot),
+  );
+
+  const [snapshotText, setSnapshotText] = useState(snapshot.toString());
+
+  const [annotationType, setMetric] = useState(
```

Comment:
Can we reconcile the naming of `annotationType` and `setMetric`?

Reply:
oops, yeah good catch